### DISCUSSION TITLE
Ladeprogramm aus der Oberfläche steuern!

### DISCUSSION THREAD

**Discussion by wiggal** [wiggal/GEN24_Ladesteuerung#69] [Ideas]
Hallo,
ich möchte eine Steuerung des Ladeprogramms in die Oberfläche integrieren.

Anwendungsfall:
Da man nun ja über das Batteriemanagement des Wechselrichters steuern kann, kommt es eventuell vor, dass man die Ladesteuerung manuelle einstellen will. Damit die Einstellung nicht gleich wieder vom Ladeprogramm überschrieben wird.

Hier mal ein Foto vom Prototypen:
![grafik](https://github.com/wiggal/GEN24_Ladesteuerung/assets/69506553/a8ea40fc-6418-4fdf-9877-06c028b59443)

Fallen euch vielleicht bessere Bezeichnungen und noch benötigte Kombinationen ein?

Würde mich über Kommentare freuen
schönen Gruß

**Comment by JauntyJosef** [comment]
Hallo Wiggal!
Ich finde die Idee gut. Ich habe diese Möglichkeit in meinem (Test)Fork ebenfalls integriert.
![grafik](https://github.com/wiggal/GEN24_Ladesteuerung/assets/64742105/ff8cea55-ec31-429f-92c2-9b623514ce8a)

Ich persönlich finde, dass "Aufrufparameter belassen" etwas unklar und nicht selbsterklärend sind.

LG

**Reply by wiggal** [reply]
OK ich habs überarbeitet:
![grafik](https://github.com/wiggal/GEN24_Ladesteuerung/assets/69506553/1d7242d4-4ac5-40fc-8d6c-c527ce6eb705)

**Comment by JauntyJosef** [comment]
Was ist der Unterschied zwischen AUS und "unverändert lassen"?

**Reply by wiggal** [reply]
Hi,
bei "unverändert lassen" läuft das Programm wie gestartet wurde (logging, schreiben oder nur Analyse).
Bei AUS wird das Programm sofort beendet, ohne Analyse usw., dann steht im Protokoll nur Date und "Programm wurde beendet" oder so ähnlich.

Wenn du bessere Begriffe weißt, immer her damit.

**Reply by JauntyJosef** [reply]
Guten Morgen!

Nur ein paar Gedanken von mir. Es soll wirklich keine Kritik sein!
Also, falls ich es richtig verstanden habe:

**Durch die Optionen auf dem Tab, können die SymoGen24Controller-Skripte gesteuert werden:
unverändert lassen -->> die Skripte laufen mit den Parametern, mit denen die aufgerufen wurden (iz.B.: loggin, schreiben)**
Soweit klar für mich.

**AUS -->> die Skripte werden ohne egliche Funktion beendet**
Verstehe ich auch, allerdings steht in der Überschrift "Ladesteuerung". Tatsächlich wird aber gar nichts augeführt.

**Analyse in Crontab.loga -->> die Skripte berechnen die Ladewerte geben die Ergebnisse nur aus.**
Wäre also so, wie wenn man das Skript mit dem Parameter "logging" aufruft, allerdings logging in der config.ini **deaktiviert** ist.

**NUR Logging -->> die Skripte führen nur das Logging für die Diagramme durch**
Wäre also so, wie wenn man das Skript mit dem Parameter "logging" aufruft, allerdings logging in der config.ini **aktivert** ist.

**WR-Steuerung und Logging -->> die Skripte schreiben die Ladesteuerung auf den WR und machen das Logging
Damit kann die Steuerung auch mal abgeschaltet werden, ohne das Logging zu unterbrechen usw.**
Also das Skript mit mit dem Parameter schreiben aufgerufen und logging in der config.ini aktiviert.

Auf der Settings Seite steht:
**ACHTUNG: Diese Einstellungen überschreiben teilweise den Aufrufparameter (z.B. vom Cronjob)!**
Was wird jetzt bei welcher Option genau überschrieben? Was ist zB bei "NUR Logging", wenn in der config.ini logging deaktiviert ist?

LG

**Reply by wiggal** [reply]
> 
> Nur ein paar Gedanken von mir. Es soll wirklich keine Kritik sein! Also, falls ich es richtig verstanden habe:

Keine Angst, verstehe die Gedanken immer als Anregungen, nicht als Kritik 

> **Durch die Optionen auf dem Tab, können die SymoGen24Controller-Skripte gesteuert werden: unverändert lassen -->> die Skripte laufen mit den Parametern, mit denen die aufgerufen wurden (z.B.: loggin, schreiben)** Soweit klar für mich.

OK 

> **AUS -->> die Skripte werden ohne egliche Funktion beendet** Verstehe ich auch, allerdings steht in der Überschrift "Ladesteuerung". Tatsächlich wird aber gar nichts augeführt.

Ja gut, die Ladesteuerung wird halt für den Zeitraum in dem die Option gewählt wurde ohne Funktion ausgeführt 

> **Analyse in Crontab.loga -->> die Skripte berechnen die Ladewerte geben die Ergebnisse nur aus.** Wäre also so, wie wenn man das Skript mit dem Parameter "logging" aufruft, allerdings logging in der config.ini **deaktiviert** ist.

Analyse hat nichts mit dem Logging zu tun, es wird die Ladeberechnung durchgeführt und alle Werte werden in der Crontab.log geschrieben, mehr nicht. Kein Logging, kein schreiben auf den WR, funktioniert genau so als wurde das Skript ohne Parameter aufgerufen.

> **NUR Logging -->> die Skripte führen nur das Logging für die Diagramme durch** Wäre also so, wie wenn man das Skript mit dem Parameter "logging" aufruft, allerdings logging in der config.ini **aktivert** ist.

Genau, das Skipt läuft hier mit dem Parameter "logging"

> **WR-Steuerung und Logging -->> die Skripte schreiben die Ladesteuerung auf den WR und machen das Logging Damit kann die Steuerung auch mal abgeschaltet werden, ohne das Logging zu unterbrechen usw.** Also das Skript mit mit dem Parameter schreiben aufgerufen und logging in der config.ini aktiviert.

Das ` Damit kann die Steuerung auch mal abgeschaltet werden, ohne das Logging zu unterbrechen usw.` ist in der Beschreibung an der falschen Stelle, sollte nur so sein:
`**WR-Steuerung und Logging -->> die Skripte schreiben die Ladesteuerung auf den WR und machen das Logging`

Das Skript wird hier einfach mit dem Parameter `schreiben` aufgerufen.
 
> Auf der Settings Seite steht: **ACHTUNG: Diese Einstellungen überschreiben teilweise den Aufrufparameter (z.B. vom Cronjob)!** Was wird jetzt bei welcher Option genau überschrieben? Was ist zB bei "NUR Logging", wenn in der config.ini logging deaktiviert ist?

Außer bei dem Punkt `unverändert lassen` wird der ürsprüngliche Parameter immer ersetzt.
Ist in der config.ini logging deaktiviert, wir das Logging nie durchgeführt, egal welche Parameter oder Optionen man verwendet.

Ich hoffe es ist ein bisschen klarer geworden.

Schönen Gruß
> 
> LG
